domain/application/service/unit.go

```diff
@@ -189,7 +211,44 @@ func (s *Service) makeIAASUnitArgs(
 			return nil, errors.Errorf("invalid placement: %w", err)
 		}
 
-		unitStorageArgs, err := makeUnitStorageArgs(storageDirectives, nil)
+		var (
+			machineUUID        coremachine.UUID
+			machineNetNodeUUID network.NetNodeUUID
+		)
+		// If the placement of the unit is on to an already established machine
+		// we need to resolve this to a machine uuid and netnode uuid.
+		if placement.Type == deployment.PlacementTypeMachine {
+			var err error
+			machineUUID, machineNetNodeUUID, err =
+				s.st.GetMachineUUIDAndNetNodeForName(
+					ctx, placement.Directive,
+				)
+			if err != nil {
+				return nil, errors.Errorf(
+					"getting machine %q for unit placement directive: %w",
+					placement.Directive, err,
+				)
+			}
```

Comment:
Only a nit pick but I think this can be cleaner by just making some temporaries.
```suggestion
			mUUID, nnUUID, err := s.st.GetMachineUUIDAndNetNodeForName(
				ctx, placement.Directive,
			)
			if err != nil {
				return nil, errors.Errorf(
					"getting machine %q for unit placement directive: %w",
					placement.Directive, err,
				)
			}
			machineUUID = mUUID
			machineNetNodeUUID = nnUUID
```